packages/pharos/src/components/table/pharos-table.ts

```diff
@@ -122,39 +140,27 @@ export class PharosTable extends ScopedRegistryMixin(PharosElement) {
     this.totalResults = !this.totalResults ? this.rowData.length : this.totalResults;
   }
 
-  private _toggleActiveStickyHeader = (active: boolean) => {
-    const ACTIVE_HEADER_CLASS = 'table-sticky-header--is-active';
-    const ACTIVE_HEADER_CELL_CLASS = 'table-sticky-header__cell--is-active';
-    const headerCells = this.header?.querySelectorAll('.table-header__cell');
-
-    if (active) {
-      this.header?.classList.add(ACTIVE_HEADER_CLASS);
-      headerCells?.forEach((cell) => {
-        cell.classList.add(ACTIVE_HEADER_CELL_CLASS);
-      });
-    } else {
-      this.header?.classList.remove(ACTIVE_HEADER_CLASS);
-      headerCells?.forEach((cell) => {
-        cell.classList.remove(ACTIVE_HEADER_CELL_CLASS);
-      });
+  private _toggleElevatedStickyHeader = (elevated: boolean) => {
+    if (this.header) {
+      this.header.elevated = elevated;
     }
   };
 
-  private _initHeaderObserver(): void {
+  private _initHeaderObserver(height: number): void {
     if (!this.header) {
       throw new Error('No table header found, cannot initialize observer');
     }
 
     this.observer = new IntersectionObserver(
       (entries: IntersectionObserverEntry[]) => {
         entries.forEach((entry) => {
-          this._toggleActiveStickyHeader(entry.isIntersecting);
+          this._toggleElevatedStickyHeader(entry.isIntersecting);
         });
       },
       {
         root: this.shadowRoot?.querySelector('.table'),
         // Negative top rootMargin to offset the viewbox used for intersection calculations by the height of the current header
-        rootMargin: `-${this.header.getBoundingClientRect().height}px 0px 0px 0px`,
+        rootMargin: `-${height}px 0px 0px 0px`,
```

Comment:
With slotted content, the header doesn't have a height until all its child elements have rendered, so we need to pass it in to be sure it isn't 0